rigid_transformations.py

```diff
@@ -0,0 +1,611 @@
+
+    Formula:
+    TODO(nina).
+
+    :param transfo: 6d vector element in SE(3)
+    :returns inverse_transfo: 6d vector inverse of transfo
+    """
+
+    transfo = regularize_transformation(transfo)
+
+    rot_vec = transfo[0:3]
+    translation = transfo[3:6]
+
+    inverse_transfo = np.zeros(6)
+    inverse_transfo[0:3] = -rot_vec
+    rot_mat = rotations.rotation_matrix_from_rotation_vector(-rot_vec)
+    inverse_transfo[3:6] = np.dot(rot_mat, -translation)
+
+    inverse_transfo = regularize_transformation(inverse_transfo)
+    return inverse_transfo
+
+
+def compose(transfo_1, transfo_2):
+    """
+    Compose two elements of group SE(3).
+
+    Formula:
+    transfo_1 . transfo_2 = [R1 * R2, (R1 * t2) + t1]
+    where:
+    R1, R2 are rotation matrices,
+    t1, t2 are translation vectors.
+
+    :param transfo_1, transfo_2: 6d vectors elements of SE(3)
+    :returns prod_transfo: composition of transfo_1 and transfo_2
+    """
+    rot_mat_1 = rotations.rotation_matrix_from_rotation_vector(transfo_1[0:3])
+    rot_mat_1 = rotations.closest_rotation_matrix(rot_mat_1)
+
+    rot_mat_2 = rotations.rotation_matrix_from_rotation_vector(transfo_1[0:3])
+    rot_mat_2 = rotations.closest_rotation_matrix(rot_mat_2)
+
+    translation_1 = transfo_1[3:6]
+    translation_2 = transfo_2[3:6]
+
+    prod_transfo = np.zeros(6)
+    prod_rot_mat = np.dot(rot_mat_1, rot_mat_2)
+
+    prod_transfo[0:3] = rotations.rotation_vector_from_rotation_matrix(
+            prod_rot_mat)
+    prod_transfo[3:6] = np.dot(rot_mat_1, translation_2) + translation_1
+
+    prod_transfo = regularize_transformation(prod_transfo)
+    return prod_transfo
+
+
+def jacobian_translation(transfo, left_or_right='left'):
+    """
+    Compute the jacobian matrix of the differential
+    of the left/right translations
+    from the identity to transfo in the Lie group SE(3).
+
+    Formula:
+    TODO(nina).
+
+    :param transfo: 6D vector element of SE(3)
+    :returns jacobian: 6x6 matrix
+    """
+    assert len(transfo) == 6
+    transfo = regularize_transformation(transfo)
+
+    rot_vec = transfo[0:3]
+    translation = transfo[3:6]
+
+    jacobian = np.zeros((6, 6))
+
+    if left_or_right == 'left':
+        jacobian_rot = rotations.jacobian_translation(rot_vec,
+                                                      left_or_right='left')
+        jacobian_trans = rotations.rotation_matrix_from_rotation_vector(
+                rot_vec)
+
+        jacobian[:3, :3] = jacobian_rot
+        jacobian[3:, 3:] = jacobian_trans
+
+    elif left_or_right == 'right':
+        jacobian_rot = rotations.jacobian_translation(rot_vec,
+                                                      left_or_right='right')
+        jacobian[:3, :3] = jacobian_rot
+        jacobian[3:, :3] = -rotations.skew_matrix_from_vector(translation)
+        jacobian[3:, 3:] = np.eye(3)
+
+    else:
+        raise ValueError('\'left_or_right\' needs \'left\' or \'right\'')
```

Comment:
you should probably do the argument checking at the beginning: 

assert left_or_right in ('left', 'right')